### PR TITLE
Fix -Wunreachable-code-aggressive in json_reader.cc

### PR DESCRIPTION
**Pull request by nico**
Bug: chromium:1066980

@veblush

### CODE REVIEW

**Comment by veblush** [comment]
@nico Can you sign EasyCLA? and please consider adding this new warning flag to https://github.com/grpc/grpc/blob/master/bazel/copts.bzl

**Comment by drfloob** [comment]
> please consider adding this new warning flag to https://github.com/grpc/grpc/blob/master/bazel/copts.bzl

Unfortunately that won't be easy. See my attempt here: https://github.com/grpc/grpc/pull/26951. It was decided that the things we'd need to change to enable this flag would leave the codebase in a worse state.

**Comment by drfloob** [comment]
That said, I neglected to land a few easy cleanup spots from that bug. I'll create a PR for just the cleanup.

**Comment by drfloob** [comment]
https://github.com/grpc/grpc/pull/27054: the cleanup for a handful of other unreachable code problems. I removed the fix for `json_reader.cc` from my PR, @nico please feel free to continue with this PR as well.

**Comment by nico** [comment]
I tried to complete the CLA form, but nothing has happened yet.

FWIW, I show up as CLA-approved on other projects (e.g. here https://github.com/google/benchmark/pull/1214 , click "show all checks").

@drfloob Please feel free to keep this in your patch too, in case the CLA issue isn't resolved soon.

**Comment by nico** [comment]
Looks like EasyCLA is happy now.

File: src/core/lib/json/json_reader.cc [hunk 1]
```diff
@@ -279,10 +279,7 @@ JsonReader::Status JsonReader::Run() {
       case GRPC_JSON_READ_CHAR_EOF:
         if (IsComplete()) {
           return Status::GRPC_JSON_DONE;
-        } else {
-          return Status::GRPC_JSON_PARSE_ERROR;
-        }
-        break;
+        return Status::GRPC_JSON_PARSE_ERROR;
```
Comment (drfloob) [inline src/core/lib/json/json_reader.cc:282]: Missing a `}`.

Comment (drfloob) [reply src/core/lib/json/json_reader.cc:282]: Alternatively, you could write `return IsComplete() ? Status::GRPC_JSON_DONE : Status::GRPC_JSON_PARSE_ERROR;`

Comment (nico) [reply src/core/lib/json/json_reader.cc:283]: Thanks, fixed.

🤦‍♂️

**Comment by drfloob** [comment]
The `Artifact Build MacOS` failure looks like an error in the ruby build. Reported https://github.com/grpc/grpc/issues/27064.

**Review by drfloob — APPROVED** [review]

**Comment by nico** [comment]
Thanks!

**Pull request merged**